src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2085,6 +2097,11 @@ SubquerySqlTaskList(Job *job)
 			}
 		}
 
+		if (anchorRangeTableId == 0 || tableSize > maxTableSize){
```

Comment:
Why don't we move `tableSize += ShardLength(shardInterval->shardId);` line to here and add an explanatory comment of what we're doing here?